lib/ansible/plugins/lookup/password.py

```diff
@@ -95,7 +96,7 @@ def random_password(self, length=DEFAULT_LENGTH, chars=C.DEFAULT_PASSWORD_CHARS)
 
         password = []
         while len(password) < length:
-            new_char = os.urandom(1)
+            new_char = to_native(os.urandom(1))
```

Comment:
We should use to_text here instead of to_native since it's controller code.  Probably need to change the default in constants.DEFAULT_PASSWORD_CHARS to be a text literal string as well.

Not new with your code but ```os.urandom(1)``` won't work correctly with non-ascii characters.  Previous code may have been worse (I think it could have generated mangled, invalid utf-8.  The new code will only omit any non-ascii chars.)

We could use ```random.SystemRandom().choice(chars)``` here.  The upstream documentation is unclear but I think that is available everywhere that os.urandom() is so we don't lose any portability by using it.